src/components/file-input/file-reader.tsx

```diff
@@ -0,0 +1,18 @@
+export function readFile(file: File) {
+  return readTxtFile(file);
+}
+
+function readTxtFile(txtFile: File) {
+  return new Promise<string>((resolve, reject) => {
+    const fileReader = new FileReader();
+    fileReader.addEventListener('load', () => {
```

Comment:
consider adding an event listener for `error` (or using .onerror) to reject the promise; I'm fairly certain `readAsText` doesn't throw any synchronous exceptions

Reply:
haha, whoops. Good catch. I got that updated